bounter/count_min_sketch.py

```diff
@@ -62,39 +52,42 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
 
         Args:
             size_mb (int): controls the maximum size of the Count-min Sketch table.
-                If width is provided, this parameter is ignored. Please note that the structure will use an
-                overhead of approximately 33 KB in addition to the table size.
+                If both width and depth is provided, this parameter is ignored.
+                Please note that the structure will use an overhead of approximately 65 KB in addition to the table size.
             depth (int): controls the number of rows of the table. Having more rows decreases probability of a big
-                overestimation but also linearly affects performance and table size. Choose a small number such as 4-8.
-                The algorithm will default depth 8 if width is provided. Otherwise, it will choose a depth in range 4-7
-                to best fill the maximum memory (for memory size which is a power of 2, depth of 4 is always used).
-            width (int): controls the number of hash buckets in one row of the table, overrides size_mb parameter.
+                overestimation but also linearly affects performance and table size. Choose a small number such as 6-10.
+                The algorithm will default depth 8 if width is provided. Otherwise, it will choose a depth in range 5-8
+                to best fill the maximum memory (for memory size which is a power of 2, depth of 8 is always used).
+            width (int): controls the number of hash buckets in one row of the table.
                 If width is not provided, the algorithm chooses the maximum width to fill the available size.
                 The more, the better, should be very large, preferably in the same order of magnitude as the cardinality
                 of the counted set.
             algorithm (str): controls the algorithm used for counting
-                - 'basic'
                 - 'conservative' (default)
-                - 'logcounter8'
-                - 'logcounter1024'
-                - 'logcons1024'
+                - 'log1024'
+                - 'log8'
         """
-        self.array_type = CountMinSketch.cell_type(algorithm)
-        cell_size = self.array_type().itemsize
+
+        cell_size = CountMinSketch.cell_size(algorithm);
+        self.cell_size_v = cell_size
```

Comment:
For an open source library, it would be better not expose the internals and prevent the user from changing the values of `self.cell_size_v`, `self.width`, `self.depth`, etc. This is an elegant solution that doesn't require much effort: https://github.com/oz123/read-only-properties

Reply:
I disagree here. We could maybe prefix `_` to mark a property as "not public".

Any other obstructions look unpythonic, I wouldn't like to see that.

Reply:
There's value in keeping the values available for reading in the public API, so _ prefix not ideal. Keeping them available for writing also not ideal, though adequate if the class is only to be used internally ("Python programmers should act as responsible adults"). For open source library, the user interface deserves more polishing, IMO.

I don't agree that making the value read-only would be an unpythonic obstruction. The `@property` annotation is Python core so that's hardly unpythonic.

Reply:
I'd rather avoid adding a package dependency for this. I could use @property if you want this feature. What do you think?